core/net/ipv6/uip-nd6.c

```diff
@@ -662,7 +662,8 @@ rs_input(void)
         }
         if(memcmp(&nd6_opt_llao[UIP_ND6_OPT_DATA_OFFSET],
             lladdr, UIP_LLADDR_LEN) != 0) {
-          uip_ds6_nbr_t nbr_data = *nbr;
+          uip_ds6_nbr_t nbr_data;
+          nbr_data = *nbr;
```

Comment:
Ah ok then I guess the problem is not just variable initialization, but the fact the variable is a struct and we derefernece it. One thing though is that as long as we don't have a regression test to verify this, the problem might slip in again any time...